### PR TITLE
Upgrade to Ruby 2.3.0

### PR DESCRIPTION
**Pull request by swanson**
This PR bumps the default version to Ruby 2.3.0.

`.ruby-version` is now included in the repo (previously ignored) and is the single source of truth for the version. Individuals wishing to deploy/run other versions of Ruby are advised to clone the repo and modify `.ruby-version` locally.

Updates the TravisCI build matrix to remove older Rubies and start running against `ruby-head` for kicks.

(Additionally, fixed a failing JS test that seemed unrelated to Ruby upgrade)

---

I've deployed this branch to my Heroku instance and everything works as expected. Looking for a few others to run off this branch and report any issues.

cc @Koronen @geemus

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ But it does have keyboard shortcuts and was made with love!
 
 ## Installation
 
-Stringer is a Ruby (2.0.0+) app based on Sinatra, ActiveRecord, PostgreSQL, Backbone.js and DelayedJob.
+Stringer is a Ruby (2.3.0+) app based on Sinatra, ActiveRecord, PostgreSQL, Backbone.js and DelayedJob.
```
Comment (bf4) [inline README.md]: Worth having docs re ruby-version here?

**Comment by geemus** [comment]
@swanson Hey, it looks like .ruby-version is still in .gitignore and not yet in the repo, is this branch maybe missing a commit? (also possible that I'm missing something in terms of intent). Looking forward to this, thanks!

**Comment by swanson** [comment]
@geemus Doh! Sorry. Commit wasn't pushed to remote :)

**Comment by geemus** [comment]
No worries, I figured it was something simple like that. Will try it out a bit later today and let you know. Thanks!

**Comment by geemus** [comment]
@swanson just deployed to heroku and checked, everything looks good, thanks!

File: README.md (hunk 1 above)
Comment (Koronen) [inline README.md]: I think so, yes.

**Comment by Koronen** [comment]
I've had my instance running on Ruby 2.3.0 for a while now, using a different branch, without any noticeable issues. I just deployed this branch to see if it's any different.

File: README.md (hunk 1 above)
Comment (swanson) [inline README.md]: Let's wait on an issue to doc this more formally.

**Pull request merged**